protocols/v2/binary-sv2/no-serde-sv2/codec/src/datatypes/non_copy_data_types/seq_inner.rs

```diff
@@ -321,3 +343,75 @@ impl<'a, const ISFIXED: bool, const SIZE: usize, const HEADERSIZE: usize, const
         Seq064K::new(static_seq).unwrap()
     }
 }
+
+/// The liftime is here only for type compatibility with serde-sv2
+#[repr(C)]
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct Sv2Option<'a, T>(pub Vec<T>, PhantomData<&'a T>);
+
+// TODO add test for that and implement it also with serde!!!!
+impl<'a, const SIZE: usize> Sv2Option<'a, super::inner::Inner<'a, true, SIZE, 0, 0>> {
+    pub fn to_option(&self) -> Option<Vec<u8>> {
```

Comment:
is there a reason not to consume self here?

Reply:
Yes. The `to` prefix in a function eg `to_option` usually in rust mean that we take a reference. Here we take a reference cause is more likely  to have a reference when we need to use it (we usually have views on byte arrays that rapresent the received message).

Reply:
its not letting me resolve so I will give a thumbs up for resolved comments